WordPress/Classes/ViewRelated/Menus/MenusViewController.m

```diff
@@ -0,0 +1,945 @@
+    CGRect visibleContentRect = [self.scrollView.window convertRect:self.scrollView.frame fromView:self.view];
+    
+    CGPoint offset = self.scrollView.contentOffset;
+    
+    visibleContentRect.origin.y += offset.y;
+    viewRectWithinScrollViewWindow.origin.y += offset.y;
+    
+    BOOL updated = NO;
+    
+    if (viewRectWithinScrollViewWindow.origin.y < visibleContentRect.origin.y + padding) {
+        
+        offset.y -= viewRectWithinScrollViewWindow.size.height;
+        updated = YES;
+        
+    } else  if (viewRectWithinScrollViewWindow.origin.y + viewRectWithinScrollViewWindow.size.height > (visibleContentRect.origin.y + visibleContentRect.size.height) - padding) {
+        offset.y += viewRectWithinScrollViewWindow.size.height;
+        updated = YES;
+    }
+    
+    if (updated) {
+        [UIView animateWithDuration:0.25 animations:^{
+            self.scrollView.contentOffset = offset;
+        }];
+    }
+}
+
+- (void)itemsViewAnimatingContentSizeChanges:(MenuItemsViewController *)itemsView focusedRect:(CGRect)focusedRect updatedFocusRect:(CGRect)updatedFocusRect
+{
+    CGPoint offset = self.scrollView.contentOffset;
+    offset.y += updatedFocusRect.origin.y - focusedRect.origin.y;
+    self.scrollView.contentOffset = offset;
+}
+
+#pragma mark - Alert Handlers
+
+- (NSString *)discardChangesAlertTitle
+{
+    return NSLocalizedString(@"Unsaved Changes", @"Menus alert title for alerting the user to unsaved changes.");
+}
+
+- (void)promptForDiscardingChangesBeforeSelectingADifferentLocation:(void(^)())confirmationBlock
+                                                       cancellation:(void(^)())cancellationBlock
+{
+    NSString *title = [self discardChangesAlertTitle];
+    NSString *message = NSLocalizedString(@"Selecting a different menu location will discard changes you've made to the current menu. Are you sure you want to continue?", @"Menus alert message for alerting the user to unsaved changes while trying to select a different menu location.");
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
+    {
+        UIAlertAction *action = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel and Keep Changes", @"Menus alert button title to cancel discarding changes and not select a new menu location")
+                                                         style:UIAlertActionStyleCancel
+                                                       handler:^(UIAlertAction * _Nonnull action) {
+                                                           if (cancellationBlock) {
+                                                               cancellationBlock();
+                                                           }
+                                                       }];
+        [alert addAction:action];
```

Comment:
FWIW there is is a helper class `UIAlertController+Helpers.swift` with some convenience methods like `addCancelActionWithTitle` 
Just a thought.